File: rb/features.rb
# Makeup SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module MakeupFeatures
  def self.make_feature(name)
    case name
    when "base"
      MakeupBaseFeature.new
    when "ratelimit"
      MakeupRatelimitFeature.new
    when "retry"
      MakeupRetryFeature.new
    when "test"
      MakeupTestFeature.new
    when "timeout"
      MakeupTimeoutFeature.new
    else
      MakeupBaseFeature.new
    end
  end
end

File: py/features.py
# Makeup SDK feature factory

from feature.base_feature import MakeupBaseFeature
from feature.test_feature import MakeupTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MakeupBaseFeature(),
        "test": lambda: MakeupTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
